src/main/java/org/kizombadev/markdownparser/HtmlGenerator.java

```diff
@@ -0,0 +1,51 @@
+package org.kizombadev.markdownparser;
+
+import org.kizombadev.markdownparser.entities.*;
+import org.kizombadev.markdownparser.entities.interfaces.ImmutableSyntax;
+
+public class HtmlGenerator {
+
+    private ImmutableSyntax root;
+
+    private HtmlGenerator() {
+    }
+
+    public static HtmlGenerator create() {
+        return new HtmlGenerator();
+    }
+
+    public String parse(ImmutableSyntax root) {
+        this.root = root;
+
+        StringBuilder html = new StringBuilder();
+
+        handle(root, html);
+
+        return html.toString();
+    }
+
+    private void handle(ImmutableSyntax root, StringBuilder html) {
+        for (ImmutableSyntax child : root.getChildren()) {
+
+            if (child instanceof BigHeadline) {
+                handleTag("h1", child, html);
+            } else if (child instanceof SmallHeadline) {
+                handleTag("h2", child, html);
+            } else if (child instanceof TextSyntax) {
+                html.append(child.convertTo(TextSyntax.class).getText());
+            } else if (child instanceof BoldSyntax) {
+                handleTag("b", child, html);
+            } else if (child instanceof ItalicSyntax) {
+                handleTag("i", child, html);
+            } else if (child instanceof ItalicSyntax) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This branch can not be reached because the condition duplicates a previous condition in the same sequence of "if/else if" statements [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1862)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 36. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)